authoring-recipes/modifying-methods-with-javatemplate.md

```diff
@@ -273,6 +273,8 @@ public JavaIsoVisitor<ExecutionContext> getVisitor() {
 
 {% hint style="info" %}
 Note: When building a template, if you use a type that's not a base Java type, you will need to specify what package that type comes from. In our case, since we're adding a `Date` to the `MethodDeclaration`, we need to specify that this is a `java.util.Date` and not some other type of date. You can do that by adding an `imports` function with the packages for the types used in the template.
+
+If your import is _not_ part of the JDK itself, you will need to specify a classpath so that the recipe will know where the function you're adding is coming from. You can do this via the `classpathFromResources` method such as in the [Migrate Hamcrest assertThat recipe](https://github.com/openrewrite/rewrite-testing-frameworks/blob/v2.0.9/src/main/java/org/openrewrite/java/testing/hamcrest/AssertThatBooleanToAssertJ.java#L55).
```

Comment:
The `.classpathFromResources(...)` is slightly more advanced than the regular `.classpath(...)`, and only necessary when using multiple different versions of a dependency. I'd suggest here we use `.classpath(...)` with this example link: https://github.com/openrewrite/rewrite-static-analysis/blob/v1.0.4/src/test/java/org/openrewrite/staticanalysis/RemoveUnneededAssertionTest.java#L105
That allows users to use "regular" dependencies instead of the `recipeDependencies { parserClasspath(...) }` dependencies.

And then separately link to https://docs.openrewrite.org/authoring-recipes/multiple-versions for those who need more than `.classpath(...)` offers.